anta/tests/routing/ospf.py

```diff
@@ -186,23 +140,26 @@ class VerifyOSPFMaxLSA(AntaTest):
     ```
     """
 
-    description = "Verifies all OSPF instances did not cross the maximum LSA threshold."
     categories: ClassVar[list[str]] = ["ospf"]
     commands: ClassVar[list[AntaCommand | AntaTemplate]] = [AntaCommand(command="show ip ospf", revision=1)]
 
     @AntaTest.anta_test
     def test(self) -> None:
         """Main test function for VerifyOSPFMaxLSA."""
-        command_output = self.instance_commands[0].json_output
-        ospf_instance_info = _get_ospf_max_lsa_info(command_output)
-        if not ospf_instance_info:
-            self.result.is_skipped("No OSPF instance found.")
+        self.result.is_success()
+
+        # If OSPF is not configured on device, test fails.
+        if not (command_output := get_value(self.instance_commands[0].json_output, "vrfs")):
+            self.result.is_skipped("OSPF not configured")
             return
-        all_instances_within_threshold = all(instance["numLsa"] <= instance["maxLsa"] * (instance["maxLsaThreshold"] / 100) for instance in ospf_instance_info)
-        if all_instances_within_threshold:
-            self.result.is_success()
-        else:
-            exceeded_instances = [
-                instance["instance"] for instance in ospf_instance_info if instance["numLsa"] > instance["maxLsa"] * (instance["maxLsaThreshold"] / 100)
-            ]
-            self.result.is_failure(f"OSPF Instances {exceeded_instances} crossed the maximum LSA threshold.")
+
+        exceeded_instances = []
+        for vrf_data in command_output.values():
+            for instance, instance_data in vrf_data.get("instList", {}).items():
+                max_lsa = instance_data.get("maxLsaInformation", {}).get("maxLsa")
+                max_lsa_threshold = instance_data.get("maxLsaInformation", {}).get("maxLsaThreshold")
+                num_lsa = instance_data.get("lsaInformation", {}).get("numLsa")
+                if num_lsa > max_lsa * (max_lsa_threshold / 100):
+                    exceeded_instances.append(instance)
+        if exceeded_instances:
+            self.result.is_failure(f"Following OSPF Instances crossed the maximum LSA threshold - {', '.join(exceeded_instances)}")
```

Comment:
```suggestion
            self.result.is_failure(f"Following OSPF instances crossed the maximum LSA threshold - {', '.join(exceeded_instances)}")
```
Can we add some numbers here? Expected Actual?